src/devices/sisyfos.ts

```diff
@@ -237,30 +263,78 @@ export class SisyfosMessageDevice extends DeviceWithState<SisyfosState> implemen
 
 			// if the tlObj is specifies to load to PST the original Layer is used to resolve the mapping
 			if (!foundMapping && layer.isLookahead && layer.lookaheadForLayer) {
-				foundMapping = this.getMapping()[layer.lookaheadForLayer] as any
+				foundMapping = mappings[layer.lookaheadForLayer] as any
 			}
-			if (foundMapping && _.has(foundMapping, 'channel') && deviceState.channels[foundMapping.channel]) {
 
-				if (layer.isLookahead) {
-					deviceState.channels[foundMapping.channel].pstOn = layer.content.isPgm || 0
-				} else {
-					deviceState.channels[foundMapping.channel].pgmOn = layer.content.isPgm || 0
+			// Preparation: put all channels that comes from the state in an array:
+			const newChannels: ({
+				overridePriority: number,
+				channel: number,
+				isLookahead: boolean
+				tlObjId: string
+			} & SisyfosChannelOptions)[] = []
+			if (foundMapping) {
+				// @ts-ignore backwards-compatibility:
+				if (!foundMapping.mappingType) foundMapping.mappingType = MappingSisyfosType.CHANNEL
+				// @ts-ignore backwards-compatibility:
+				if (content.type === 'sisyfos') content.type = TimelineContentTypeSisyfos.CHANNEL
+
+				if (
+					foundMapping.mappingType === MappingSisyfosType.CHANNEL &&
+					content.type === TimelineContentTypeSisyfos.CHANNEL
+				) {
+					newChannels.push({
+						...content,
+						channel: foundMapping.channel,
+						overridePriority: content.overridePriority || 0,
+						isLookahead: layer.isLookahead || false,
+						tlObjId: layer.id
+					})
+				} else if (
+					foundMapping.mappingType === MappingSisyfosType.CHANNELS &&
+					content.type === TimelineContentTypeSisyfos.CHANNELS
+				) {
+					_.each(content.channels, channel => {
```

Comment:
Still using underscore? :) `Object.entries(content.channels).forEach([i, channel] => { ... })` achieves the same thing

Reply:
Or `for (const [i, channel] of Object.entries(content.channels)) {` to get rid of the extra layer of function and potential context switch

Reply:
true, let's keep that in mind for when we decide to ditch it ;)

Reply:
Yeah, but it would be easier to gradually replace uses whenever we work nearby. Besides, we may not be fully remove it, as there are some useful utilities